tree/dataframe/test/dataframe_interface.cxx

```diff
@@ -776,3 +776,33 @@ TEST(RDataFrameInterface, SnapshotWithDuplicateColumns)
       std::logic_error);
    EXPECT_THROW((ROOT::RDataFrame(1).Snapshot("t", "neverwritten.root", {"rdfentry_", "rdfentry_"})), std::logic_error);
 }
+
+struct Jet {
+   double a, b;
+};
+
+struct Boo {
+   TH2D h{"", "", 10, 0, 10, 10, 0, 10};
+
+   void Fill(const Jet &j) { h.Fill(j.a, j.b); }
+
+   void Merge(const std::vector<Boo *> &others)
+   {
+      TList l;
+      for (auto *o : others) {
+         l.Add(&o->h);
+         h.Merge(&l);
```

Comment:
Doesn't this merge in the first iteration `others[0]`, in the second `others[1]` and again `others[0]`, etc? Should the test detect this?

Reply:
yikes, thanks. the Merge is only used for multi-thread runs and we are not testing that case here (we are not interested in the correctness of the helper, rather in checking that RDF can use helpers with arbitrary `Fill` signatures). I'll just leave `Merge` empty.